### PR TITLE
[SPARK-11505][SQL] Break aggregate functions into multiple files

### PR DESCRIPTION
**Pull request by rxin**
functions.scala was getting pretty long. I broke it into multiple files.

I also added explicit data types for some public vals, and renamed aggregate function pretty names to lower case, which is more consistent with rest of the functions.

### CODE REVIEW

**Comment by yhuai** [comment]
LGTM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Variance.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    inputAggBufferOffset: Int = 0)
+  extends CentralMomentAgg(child) {
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): ImperativeAggregate =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): ImperativeAggregate =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override def prettyName: String = "var_pop"
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Variance.scala]: cc @sethah

next time please use more consistent naming. If the SQL function is named var_pop, don't use variance_pop here.

**Comment by yhuai** [comment]
test this please

**Comment by yhuai** [comment]
Merging to master.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Variance.scala (hunk 1 above)
Comment (sethah) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Variance.scala]: Duly noted. Thanks for the feedback!

**Pull request closed without merging**